### ISSUE TITLE
core during drain nemesis

### ISSUE THREAD

**Issue by fgelcer** [scylladb/scylla#5403]
*Installation details*
Scylla version (or git commit hash): 666.development-0.20191130.96009881d8c
Cluster size: 6
OS (RHEL/CentOS/Ubuntu/AWS AMI): ami-092c239b8d07364ec

core can be found [here](https://storage.cloud.google.com/upload.scylladb.com/core.scylla.996.fed2cd902e314f78b442ff5378bdb473.1754.1575211847000000/core.scylla.996.fed2cd902e314f78b442ff5378bdb473.1754.1575211847000000.gz)

the core was dumped after a segmentation fault:
```
p:INFO  > 2019-12-01T14:50:47+00:00  ip-172-30-0-29 !INFO    | scylla: Segmentation fault on shard 4.
0x0000000002b9b392
0x0000000002b1692b
0x0000000002b16c25
0x0000000002b16c70
0x00007f1af6a6cb1f
0x0000000001441836
0x00000000019d4495
0x00000000019d582d
0x0000000000895703
0x0000000002b10370
0x0000000002b44674
0x0000000002ba60b5
0x0000000002b0ff9c
0x0000000002b7abd8
0x0000000002b8ef9c
0x0000000002af5d7d
/libreloc/libpthread.so.0+0x00000000000094e1
/libreloc/libc.so.6+0x0000000000101642
```

decoded:
```
void seastar::backtrace<seastar::backtrace_buffer::append_backtrace()::{lambda(seastar::frame)#1}>(seastar::backtrace_buffer::append_backtrace()::{lambda(seastar::frame)#1}&&) at /usr/include/fmt/format.h:1158
seastar::backtrace_buffer::append_backtrace() at /usr/include/fmt/format.h:1158
 (inlined by) print_with_backtrace at /jenkins/workspace/scylla-master/build/scylla/seastar/src/core/reactor.cc:701
seastar::print_with_backtrace(char const*) at /usr/include/fmt/format.h:1158
sigsegv_action at /usr/include/fmt/format.h:1158
 (inlined by) operator() at /jenkins/workspace/scylla-master/build/scylla/seastar/src/core/reactor.cc:3356
 (inlined by) _FUN at /jenkins/workspace/scylla-master/build/scylla/seastar/src/core/reactor.cc:3352
?? ??:0
service::migration_manager::unregister_listener(service::migration_listener*) at crtstuff.c:?
seastar::future<>::finally_body<db::view::view_builder::stop()::{lambda()#1}, true>::operator()(seastar::future<>&&) at view.cc:?
db::view::view_builder::stop() at crtstuff.c:?
seastar::smp_message_queue::async_work_item<seastar::sharded<db::view::view_builder>::stop()::{lambda(unsigned int)#1}::operator()(unsigned int)::{lambda()#1}>::process() at crtstuff.c:?
operator() at /usr/include/fmt/format.h:1158
 (inlined by) process_queue<2, seastar::smp_message_queue::process_incoming()::<lambda(seastar::smp_message_queue::work_item*)> > at /jenkins/workspace/scylla-master/build/scylla/seastar/src/core/reactor.cc:3004
seastar::smp_message_queue::process_incoming() at /usr/include/fmt/format.h:1158
 (inlined by) seastar::smp::poll_queues() at /jenkins/workspace/scylla-master/build/scylla/seastar/src/core/reactor.cc:3825
seastar::reactor::smp_pollfn::poll() at /usr/include/fmt/format.h:1158
seastar::reactor::poll_once() at /usr/include/fmt/format.h:1158
 (inlined by) operator() at /jenkins/workspace/scylla-master/build/scylla/seastar/src/core/reactor.cc:2634
 (inlined by) _M_invoke at /usr/include/c++/9/bits/std_function.h:285
std::function<bool ()>::operator()() const at /usr/include/fmt/format.h:1158
 (inlined by) seastar::reactor::run() at /jenkins/workspace/scylla-master/build/scylla/seastar/src/core/reactor.cc:2659
seastar::smp::configure(boost::program_options::variables_map, seastar::reactor_config)::{lambda()#3}::operator()() const at /usr/include/fmt/format.h:1158
std::function<void ()>::operator()() const at /usr/include/c++/9/bits/std_function.h:690
 (inlined by) seastar::posix_thread::start_routine(void*) at /jenkins/workspace/scylla-master/build/scylla/seastar/src/core/posix.cc:52
```

```
p:INFO  > backtrace=           PID: 1754 (scylla)
p:INFO  >            UID: 996 (scylla)
p:INFO  >            GID: 1001 (scylla)
p:INFO  >         Signal: 11 (SEGV)
p:INFO  >      Timestamp: Sun 2019-12-01 14:50:47 UTC (2min 46s ago)
p:INFO  >   Command Line: /usr/bin/scylla --blocked-reactor-notify-ms 500 --abort-on-lsa-bad-alloc 1 --abort-on-seastar-bad-alloc --abort-on-internal-error 1 --log-to-syslog 1 --log-to-stdout 0 --default-log-level info --network-stack posix --io-properties-file=/etc/scylla.d/io_properties.yaml --cpuset 1-7,9-15
p:INFO  >     Executable: /opt/scylladb/libexec/scylla
p:INFO  >  Control Group: /
p:INFO  >        Boot ID: fed2cd902e314f78b442ff5378bdb473
p:INFO  >     Machine ID: df877a200226bc47d06f26dae0736ec9
p:INFO  >       Hostname: ip-172-30-0-29.ec2.internal
p:INFO  >       Coredump: /var/lib/systemd/coredump/core.scylla.996.fed2cd902e314f78b442ff5378bdb473.1754.1575211847000000
p:INFO  >        Message: Process 1754 (scylla) of user 996 dumped core.
p:INFO  >                 
p:INFO  >                 Stack trace of thread 1758:
p:INFO  >                 #0  0x0000000001441837 _ZN7service17migration_manager19unregister_listenerEPNS_18migration_listenerE (scylla)
p:INFO  >                 #1  0x00000000019d4496 _ZN7seastar6futureIJEE12finally_bodyIZN2db4view12view_builder4stopEvEUlvE_Lb1EEclEOS1_ (scylla)
p:INFO  >                 #2  0x00000000019d582e _ZN2db4view12view_builder4stopEv (scylla)
p:INFO  >                 #3  0x0000000000895704 _ZN7seastar17smp_message_queue15async_work_itemIZZNS_7shardedIN2db4view12view_builderEE4stopEvENUljE_clEjEUlvE_E7processEv (scylla)
p:INFO  >                 #4  0x0000000002b10371 operator() (scylla)
p:INFO  >                 #5  0x0000000002b44675 _ZN7seastar17smp_message_queue16process_incomingEv (scylla)
p:INFO  >                 #6  0x0000000002ba60b6 _ZN7seastar7reactor10smp_pollfn4pollEv (scylla)
p:INFO  >                 #7  0x0000000002b0ff9d _ZN7seastar7reactor9poll_onceEv (scylla)
p:INFO  >                 #8  0x0000000002b7abd9 _ZNKSt8functionIFbvEEclEv (scylla)
p:INFO  >                 #9  0x0000000002b8ef9d _ZZN7seastar3smp9configureEN5boost15program_options13variables_mapENS_14reactor_configEENKUlvE1_clEv (scylla)
p:INFO  >                 #10 0x0000000002af5d7e _ZNKSt8functionIFvvEEclEv (scylla)
p:INFO  >                 #11 0x00007f1af6a614e2 start_thread (libpthread.so.0)
p:INFO  >                 #12 0x00007f1af60fd643 __clone (libc.so.6)
p:INFO  >                 
p:INFO  >                 Stack trace of thread 1779:
p:INFO  >                 #0  0x00007f1af6a6b87c read (libpthread.so.0)
p:INFO  >                 #1  0x0000000002f4d836 _ZN7seastar11thread_pool4workENS_13basic_sstringIcjLj15ELb1EEE (scylla)
p:INFO  >                 #2  0x0000000002f4db22 operator() (scylla)
p:INFO  >                 #3  0x0000000002af5d7e _ZNKSt8functionIFvvEEclEv (scylla)
p:INFO  >                 #4  0x00007f1af6a614e2 start_thread (libpthread.so.0)
p:INFO  >                 #5  0x00007f1af60fd643 __clone (libc.so.6)
p:INFO  >                 
p:INFO  >                 Stack trace of thread 1768:
p:INFO  >                 #0  0x00007f1af6a6b87c read (libpthread.so.0)
p:INFO  >                 #1  0x0000000002f4d836 _ZN7seastar11thread_pool4workENS_13basic_sstringIcjLj15ELb1EEE (scylla)
p:INFO  >                 #2  0x0000000002f4db22 operator() (scylla)
p:INFO  >                 #3  0x0000000002af5d7e _ZNKSt8functionIFvvEEclEv (scylla)
p:INFO  >                 #4  0x00007f1af6a614e2 start_thread (libpthread.so.0)
p:INFO  >                 #5  0x00007f1af60fd643 __clone (libc.so.6)
p:INFO  >                 
p:INFO  >                 Stack trace of thread 1770:
p:INFO  >                 #0  0x00007f1af6a6b87c read (libpthread.so.0)
p:INFO  >                 #1  0x0000000002f4d836 _ZN7seastar11thread_pool4workENS_13basic_sstringIcjLj15ELb1EEE (scylla)
p:INFO  >                 #2  0x0000000002f4db22 operator() (scylla)
p:INFO  >                 #3  0x0000000002af5d7e _ZNKSt8functionIFvvEEclEv (scylla)
p:INFO  >                 #4  0x00007f1af6a614e2 start_thread (libpthread.so.0)
p:INFO  >                 #5  0x00007f1af60fd643 __clone (libc.so.6)
p:INFO  >                 
p:INFO  >                 Stack trace of thread 1772:
p:INFO  >                 #0  0x00007f1af6a6b87c read (libpthread.so.0)
p:INFO  >                 #1  0x0000000002f4d836 _ZN7seastar11thread_pool4workENS_13basic_sstringIcjLj15ELb1EEE (scylla)
p:INFO  >                 #2  0x0000000002f4db22 operator() (scylla)
p:INFO  >                 #3  0x0000000002af5d7e _ZNKSt8functionIFvvEEclEv (scylla)
p:INFO  >                 #4  0x00007f1af6a614e2 start_thread (libpthread.so.0)
p:INFO  >                 #5  0x00007f1af60fd643 __clone (libc.so.6)
p:INFO  >                 
p:INFO  >                 Stack trace of thread 1773:
p:INFO  >                 #0  0x00007f1af6a6b87c read (libpthread.so.0)
p:INFO  >                 #1  0x0000000002f4d836 _ZN7seastar11thread_pool4workENS_13basic_sstringIcjLj15ELb1EEE (scylla)
p:INFO  >                 #2  0x0000000002f4db22 operator() (scylla)
p:INFO  >                 #3  0x0000000002af5d7e _ZNKSt8functionIFvvEEclEv (scylla)
p:INFO  >                 #4  0x00007f1af6a614e2 start_thread (libpthread.so.0)
p:INFO  >                 #5  0x00007f1af60fd643 __clone (libc.so.6)
p:INFO  >                 
p:INFO  >                 Stack trace of thread 1771:
p:INFO  >                 #0  0x00007f1af6a6b87c read (libpthread.so.0)
p:INFO  >                 #1  0x0000000002f4d836 _ZN7seastar11thread_pool4workENS_13basic_sstringIcjLj15ELb1EEE (scylla)
p:INFO  >                 #2  0x0000000002f4db22 operator() (scylla)
p:INFO  >                 #3  0x0000000002af5d7e _ZNKSt8functionIFvvEEclEv (scylla)
p:INFO  >                 #4  0x00007f1af6a614e2 start_thread (libpthread.so.0)
p:INFO  >                 #5  0x00007f1af60fd643 __clone (libc.so.6)
p:INFO  >                 
p:INFO  >                 Stack trace of thread 1778:
p:INFO  >                 #0  0x00007f1af6a6b87c read (libpthread.so.0)
p:INFO  >                 #1  0x0000000002f4d836 _ZN7seastar11thread_pool4workENS_13basic_sstringIcjLj15ELb1EEE (scylla)
p:INFO  >                 #2  0x0000000002f4db22 operator() (scylla)
p:INFO  >                 #3  0x0000000002af5d7e _ZNKSt8functionIFvvEEclEv (scylla)
p:INFO  >                 #4  0x00007f1af6a614e2 start_thread (libpthread.so.0)
p:INFO  >                 #5  0x00007f1af60fd643 __clone (libc.so.6)
p:INFO  >                 
p:INFO  >                 Stack trace of thread 1774:
p:INFO  >                 #0  0x00007f1af6a6b87c read (libpthread.so.0)
p:INFO  >                 #1  0x0000000002f4d836 _ZN7seastar11thread_pool4workENS_13basic_sstringIcjLj15ELb1EEE (scylla)
p:INFO  >                 #2  0x0000000002f4db22 operator() (scylla)
p:INFO  >                 #3  0x0000000002af5d7e _ZNKSt8functionIFvvEEclEv (scylla)
p:INFO  >                 #4  0x00007f1af6a614e2 start_thread (libpthread.so.0)
p:INFO  >                 #5  0x00007f1af60fd643 __clone (libc.so.6)
p:INFO  >                 
p:INFO  >                 Stack trace
p:INFO  > 
p:INFO  > download_instructions=
p:INFO  > gsutil cp gs://upload.scylladb.com/core.scylla.996.fed2cd902e314f78b442ff5378bdb473.1754.1575211847000000/core.scylla.996.fed2cd902e314f78b442ff5378bdb473.1754.1575211847000000.gz .
p:INFO  > gunzip /var/lib/systemd/coredump/core.scylla.996.fed2cd902e314f78b442ff5378bdb473.1754.1575211847000000.gz

```

**Comment by bhalevy** [comment]
@xemul please investigate core in this issue to see if it could be explained by #5404, and if so we can mark one of them as duplicate.

**Comment by juliayakovlev** [comment]
Same issue was received during longevity-10gb-3h test on master.
Scylla version: 666.development-0.20191202.28355af134f

During drain nemesis got segmentation fault:
```
2019-12-05T02:20:18+00:00  ip-172-30-0-151 !INFO    | scylla: [shard 0] storage_service - DRAINING: starting drain process
2019-12-05T02:20:18+00:00  ip-172-30-0-151 !INFO    | scylla: [shard 0] storage_service - Thrift server stopped
2019-12-05T02:20:18+00:00  ip-172-30-0-151 !INFO    | scylla: [shard 0] storage_service - CQL server stopped
2019-12-05T02:20:18+00:00  ip-172-30-0-151 !INFO    | scylla: [shard 0] gossip - My status = NORMAL
2019-12-05T02:20:18+00:00  ip-172-30-0-151 !INFO    | scylla: [shard 0] gossip - Announcing shutdown
2019-12-05T02:20:18+00:00  ip-172-30-0-151 !INFO    | scylla: [shard 0] storage_service - Node 172.30.0.151 state jump to normal
2019-12-05T02:20:18+00:00  ip-172-30-0-151 !INFO    | scylla: [shard 12] compaction - Compacting [/var/lib/scylla/data/system/peers-37f71aca7dc2383ba70672528af04d4f/mc-320-big-Data.db:level=0, /var/lib/scylla/data/system/peers-37f71aca7d
c2383ba70672528af04d4f/mc-306-big-Data.db:level=0, ]
2019-12-05T02:20:18+00:00  ip-172-30-0-151 !INFO    | scylla: [shard 12] compaction - Compacted 2 sstables to [/var/lib/scylla/data/system/peers-37f71aca7dc2383ba70672528af04d4f/mc-334-big-Data.db:level=0, ]. 11266 bytes to 5789 (~51% of
 original) in 13ms = 0.42MB/s. ~256 total partitions merged to 1.
2019-12-05T02:20:20+00:00  ip-172-30-0-151 !INFO    | scylla: [shard 0] gossip - Disable and wait for gossip loop started
2019-12-05T02:20:20+00:00  ip-172-30-0-151 !INFO    | scylla: [shard 0] gossip - Disable and wait for gossip loop finished
2019-12-05T02:20:20+00:00  ip-172-30-0-151 !INFO    | scylla: [shard 0] gossip - Gossip is now stopped
2019-12-05T02:20:20+00:00  ip-172-30-0-151 !INFO    | scylla: [shard 0] storage_service - messaging_service stopped
2019-12-05T02:20:23+00:00  ip-172-30-0-151 !INFO    | scylla: [shard 0] storage_service - DRAINED:
....................................................................................................................................................
....................................................................................................................................................
2019-12-05T02:22:05+00:00  ip-172-30-0-151 !INFO    | systemd: Stopped Run Scylla Housekeeping restart mode.
2019-12-05T02:22:05+00:00  ip-172-30-0-151 !INFO    | systemd: Stopping Scylla JMX...
2019-12-05T02:22:05+00:00  ip-172-30-0-151 !INFO    | systemd: Stopped Run Scylla Housekeeping daily mode.
2019-12-05T02:22:06+00:00  ip-172-30-0-151 !NOTICE  | systemd: scylla-jmx.service: main process exited, code=exited, status=143/n/a
2019-12-05T02:22:06+00:00  ip-172-30-0-151 !INFO    | systemd: Stopped Scylla JMX.
2019-12-05T02:22:06+00:00  ip-172-30-0-151 !NOTICE  | systemd: Unit scylla-jmx.service entered failed state.
2019-12-05T02:22:06+00:00  ip-172-30-0-151 !WARNING | systemd: scylla-jmx.service failed.
2019-12-05T02:22:06+00:00  ip-172-30-0-151 !INFO    | systemd: Stopping Scylla Server...
2019-12-05T02:22:06+00:00  ip-172-30-0-151 !INFO    | scylla: [shard 0] init - Signal received; shutting down
2019-12-05T02:22:06+00:00  ip-172-30-0-151 !INFO    | scylla: [shard 0] init - Shutting down redis service
2019-12-05T02:22:06+00:00  ip-172-30-0-151 !INFO    | scylla: [shard 0] init - Shutting down redis service was successful
2019-12-05T02:22:06+00:00  ip-172-30-0-151 !INFO    | scylla: Segmentation fault on shard 8.
2019-12-05T02:22:06+00:00  ip-172-30-0-151 !INFO    | scylla: Backtrace:
2019-12-05T02:22:06+00:00  ip-172-30-0-151 !INFO    | scylla: 0x0000000002b9fc12
2019-12-05T02:22:06+00:00  ip-172-30-0-151 !INFO    | scylla: 0x0000000002b1b1ab
2019-12-05T02:22:06+00:00  ip-172-30-0-151 !INFO    | scylla: 0x0000000002b1b4a5
2019-12-05T02:22:06+00:00  ip-172-30-0-151 !INFO    | scylla: 0x0000000002b1b4f0
2019-12-05T02:22:06+00:00  ip-172-30-0-151 !INFO    | scylla: 0x00007f745c808b1f
2019-12-05T02:22:06+00:00  ip-172-30-0-151 !INFO    | scylla: 0x0000000001443046
2019-12-05T02:22:06+00:00  ip-172-30-0-151 !INFO    | scylla: 0x00000000019d5ca5
2019-12-05T02:22:06+00:00  ip-172-30-0-151 !INFO    | scylla: 0x00000000019d703d
2019-12-05T02:22:06+00:00  ip-172-30-0-151 !INFO    | scylla: 0x0000000000896eb3
2019-12-05T02:22:06+00:00  ip-172-30-0-151 !INFO    | scylla: 0x0000000002b14bf0
2019-12-05T02:22:06+00:00  ip-172-30-0-151 !INFO    | scylla: 0x0000000002b48ef4
2019-12-05T02:22:06+00:00  ip-172-30-0-151 !INFO    | scylla: 0x0000000002baa935
2019-12-05T02:22:06+00:00  ip-172-30-0-151 !INFO    | scylla: 0x0000000002b1481c
2019-12-05T02:22:06+00:00  ip-172-30-0-151 !INFO    | scylla: 0x0000000002b7f458
2019-12-05T02:22:06+00:00  ip-172-30-0-151 !INFO    | scylla: 0x0000000002b9381c
2019-12-05T02:22:06+00:00  ip-172-30-0-151 !INFO    | scylla: 0x0000000002afa5fd
2019-12-05T02:22:06+00:00  ip-172-30-0-151 !INFO    | scylla: /opt/scylladb/libreloc/libpthread.so.0+0x00000000000094e1
2019-12-05T02:22:06+00:00  ip-172-30-0-151 !INFO    | scylla: /opt/scylladb/libreloc/libc.so.6+0x0000000000101642
```

Coredump:
```
       PID: 1754 (scylla)
           UID: 996 (scylla)
           GID: 1001 (scylla)
        Signal: 11 (SEGV)
     Timestamp: Thu 2019-12-05 02:22:06 UTC (2min 19s ago)
  Command Line: /usr/bin/scylla --blocked-reactor-notify-ms 500 --abort-on-lsa-bad-alloc 1 --abort-on-seastar-bad-alloc --abort-on-internal-error 1 --log-to-syslog 1 --log-to-stdout 0 --default-log-level info --network-stack posix --io-p
roperties-file=/etc/scylla.d/io_properties.yaml --cpuset 1-7,9-15
    Executable: /opt/scylladb/libexec/scylla
 Control Group: /
       Boot ID: 253e06034b3f44a68f2d95f884653bdc
    Machine ID: df877a200226bc47d06f26dae0736ec9
      Hostname: ip-172-30-0-151.ec2.internal
      Coredump: /var/lib/systemd/coredump/core.scylla.996.253e06034b3f44a68f2d95f884653bdc.1754.1575512526000000
       Message: Process 1754 (scylla) of user 996 dumped core.
                
                Stack trace of thread 1762:
                #0  0x0000000001443047 _ZN7service17migration_manager19unregister_listenerEPNS_18migration_listenerE (scylla)
                #1  0x00000000019d5ca6 _ZN7seastar6futureIJEE12finally_bodyIZN2db4view12view_builder4stopEvEUlvE_Lb1EEclEOS1_ (scylla)
                #2  0x00000000019d703e _ZN2db4view12view_builder4stopEv (scylla)
                #3  0x0000000000896eb4 _ZN7seastar17smp_message_queue15async_work_itemIZZNS_7shardedIN2db4view12view_builderEE4stopEvENUljE_clEjEUlvE_E7processEv (scylla)
                #4  0x0000000002b14bf1 operator() (scylla)
                #5  0x0000000002b48ef5 _ZN7seastar17smp_message_queue16process_incomingEv (scylla)
                #6  0x0000000002baa936 _ZN7seastar7reactor10smp_pollfn4pollEv (scylla)
                #7  0x0000000002b1481d _ZN7seastar7reactor9poll_onceEv (scylla)
                #8  0x0000000002b7f459 _ZNKSt8functionIFbvEEclEv (scylla)
                #9  0x0000000002b9381d _ZZN7seastar3smp9configureEN5boost15program_options13variables_mapENS_14reactor_configEENKUlvE1_clEv (scylla)
                #10 0x0000000002afa5fe _ZNKSt8functionIFvvEEclEv (scylla)
                #11 0x00007f745c7fd4e2 start_thread (libpthread.so.0)
                #12 0x00007f745be99643 __clone (libc.so.6)
                
                Stack trace of thread 1769:
                #0  0x00007f745c80787c read (libpthread.so.0)
                #1  0x0000000002f520b6 _ZN7seastar11thread_pool4workENS_13basic_sstringIcjLj15ELb1EEE (scylla)
                #2  0x0000000002f523a2 operator() (scylla)
                #3  0x0000000002afa5fe _ZNKSt8functionIFvvEEclEv (scylla)
                #4  0x00007f745c7fd4e2 start_thread (libpthread.so.0)
                #5  0x00007f745be99643 __clone (libc.so.6)
                
                Stack trace of thread 1775:
                #0  0x00007f745c80787c read (libpthread.so.0)
                #1  0x0000000002f520b6 _ZN7seastar11thread_pool4workENS_13basic_sstringIcjLj15ELb1EEE (scylla)
                #2  0x0000000002f523a2 operator() (scylla)
                #3  0x0000000002afa5fe _ZNKSt8functionIFvvEEclEv (scylla)
                #4  0x00007f745c7fd4e2 start_thread (libpthread.so.0)
                #5  0x00007f745be99643 __clone (libc.so.6)
                
                Stack trace of thread 1768:
                #0  0x00007f745c80787c read (libpthread.so.0)
                #1  0x0000000002f520b6 _ZN7seastar11thread_pool4workENS_13basic_sstringIcjLj15ELb1EEE (scylla)
                #2  0x0000000002f523a2 operator() (scylla)
               #3  0x0000000002afa5fe _ZNKSt8functionIFvvEEclEv (scylla)
                #4  0x00007f745c7fd4e2 start_thread (libpthread.so.0)
                #5  0x00007f745be99643 __clone (libc.so.6)
                
                Stack trace of thread 1780:
                #0  0x00007f745c80787c read (libpthread.so.0)
                #1  0x0000000002f520b6 _ZN7seastar11thread_pool4workENS_13basic_sstringIcjLj15ELb1EEE (scylla)
                #2  0x0000000002f523a2 operator() (scylla)
                #3  0x0000000002afa5fe _ZNKSt8functionIFvvEEclEv (scylla)
                #4  0x00007f745c7fd4e2 start_thread (libpthread.so.0)
                #5  0x00007f745be99643 __clone (libc.so.6)
                
                Stack trace of thread 1777:
                #0  0x00007f745c80787c read (libpthread.so.0)
                #1  0x0000000002f520b6 _ZN7seastar11thread_pool4workENS_13basic_sstringIcjLj15ELb1EEE (scylla)
                #2  0x0000000002f523a2 operator() (scylla)
                #3  0x0000000002afa5fe _ZNKSt8functionIFvvEEclEv (scylla)
                #4  0x00007f745c7fd4e2 start_thread (libpthread.so.0)
                #5  0x00007f745be99643 __clone (libc.so.6)
                
                Stack trace of thread 1773:
                #0  0x00007f745c80787c read (libpthread.so.0)
                #1  0x0000000002f520b6 _ZN7seastar11thread_pool4workENS_13basic_sstringIcjLj15ELb1EEE (scylla)
                #2  0x0000000002f523a2 operator() (scylla)
                #3  0x0000000002afa5fe _ZNKSt8functionIFvvEEclEv (scylla)
                #4  0x00007f745c7fd4e2 start_thread (libpthread.so.0)
                #5  0x00007f745be99643 __clone (libc.so.6)
                
                Stack trace of thread 1776:
                #0  0x00007f745c80787c read (libpthread.so.0)
                #1  0x0000000002f520b6 _ZN7seastar11thread_pool4workENS_13basic_sstringIcjLj15ELb1EEE (scylla)
                #2  0x0000000002f523a2 operator() (scylla)
                #3  0x0000000002afa5fe _ZNKSt8functionIFvvEEclEv (scylla)
                #4  0x00007f745c7fd4e2 start_thread (libpthread.so.0)
                #5  0x00007f745be99643 __clone (libc.so.6)
                
                Stack trace of thread 1774:
                #0  0x00007f745c80787c read (libpthread.so.0)
                #1  0x0000000002f520b6 _ZN7seastar11thread_pool4workENS_13basic_sstringIcjLj15ELb1EEE (scylla)
                #2  0x0000000002f523a2 operator() (scylla)
                #3  0x0000000002afa5fe _ZNKSt8functionIFvvEEclEv (scylla)
                #4  0x00007f745c7fd4e2 start_thread (libpthread.so.0)
                #5  0x00007f745be99643 __clone (libc.so.6)
```

Download instruction:
```
gsutil cp gs://upload.scylladb.com/core.scylla.996.253e06034b3f44a68f2d95f884653bdc.1754.1575512526000000/core.scylla.996.253e06034b3f44a68f2d95f884653bdc.1754.1575512526000000.gz .
gunzip /var/lib/systemd/coredump/core.scylla.996.253e06034b3f44a68f2d95f884653bdc.1754.1575512526000000.gz
```

**Comment by xemul** [comment]
@bhalevy , I've reproduced the issue locally. The problem is that on node drain the migration manager is stopped and all instances of it are released. After this on node stop the view builder is stopped and it tries to unlist itself from mm's _listeners vector. If by that time the mm's memory has not yet been re-used (which is not unlikely), the unlisting works OK. I've poisoned the _listeners in mm destructor with an ugly memset hack, and stop crashed with pretty similar backtrace.

The patchset I've sent yesterday fixes is.
